### PR TITLE
[3.2] controller::extract_chain_id_from_db() does not properly handle dirty db

### PR DESCRIPTION
**Pull request by 766C6164**
This change adds following logic:
- all exceptions  except system_error are propagated. system_error happens for example if database do not exist and needs to be created
- if chain_id cannot be retrieved empty variant returned, get replaced with find

### CODE REVIEW

File: libraries/chain/controller.cpp [hunk 1]
```diff
@@ -3476,11 +3476,14 @@ std::optional<chain_id_type> controller::extract_chain_id_from_db( const path& s
...
-      throw;
-   } catch( ... ) {
-   }
+      auto * gpo = db.find<global_property_object>();
+      if (gpo==nullptr) return {};
+
+      return gpo->chain_id;
+
+   } catch (std::system_error &) {   
+
```
Comment (heifner) [inline libraries/chain/controller.cpp:3485]: Can you add a comment about why we are ignoring `std::system_error`. Also please follow code style found in file.

**Review by heifner — CHANGES_REQUESTED** [review]

File: libraries/chain/controller.cpp (hunk 1 above)
Comment (766C6164) [reply libraries/chain/controller.cpp:3485]: > Can you add a comment about why we are ignoring `std::system_error`. Also please follow code style found in file.

Sure, this was done to handle situation when there was no database and it needs to be created (db_error_code::not_found" exception). Not sure if the right way to plug it here or fix it differently.

Comment (heifner) [reply libraries/chain/controller.cpp:3485]: Sorry, I meant to add a comment to the code.

Comment (766C6164) [reply libraries/chain/controller.cpp:3485]: Ok, got it :)

**Review by heifner — APPROVED** [review]

**Pull request merged**